File: ios/FirebaseMessagingModule.m
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(FirebaseMessagingModule, NSObject)

RCT_EXTERN_METHOD(getToken:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(eventsNotifyReady:(BOOL)ready)
RCT_EXTERN_METHOD(eventsAddListener:(NSString *)eventName)
RCT_EXTERN_METHOD(eventsRemoveListener:(NSString *)eventName all:(BOOL)all)

@end

File: ios/FirebaseMessagingModule.swift
import FirebaseMessaging

@objc(FirebaseMessagingModule)
class FirebaseMessagingModule: NSObject {
  
  @objc func getToken(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    Messaging.messaging().token { token, error in
      if let error = error {
        reject("FCM Token Error", error.localizedDescription, error)
      } else if let token = token {
        resolve(token)
      }
    }
  }
  
  @objc func eventsNotifyReady(_ ready: Bool) {
    FirebaseEventModule.shared?.notifyJsReady(ready)
  }
  
  @objc func eventsAddListener(_ eventName: String) {
    FirebaseEventModule.shared?.addListener(eventName)
  }
  
  @objc func eventsRemoveListener(_ eventName: String, all: Bool) {
    FirebaseEventModule.shared?.removeListeners(name: eventName, all: all)
  }
}
